json_schema_fuzz/__init__.py

```diff
@@ -1,15 +1,130 @@
 """JSON schema fuzzer."""
 import random
 import string
+from decimal import Decimal
 
 import exrex
 
 from .schema_operations import merge
+from .utils import (ALL_TYPES, custom_json_loads, listify,
+                    random_multiple_in_range)
+
+MAX_REJECTED_SAMPLES = 1000
+
+
+class RejectionSamplingFailed(Exception):
+    """
+    Failed to generate sample that satisfies all criteria
+    """
 
 
 def random_integer(schema):
     """Generate random integer."""
-    return random.randint(0, 10)
+
+    multiple_of = schema.get("multipleOf", Decimal(1))
+
+    default_min = Decimal(1 * multiple_of)
+    default_max = Decimal(100 * multiple_of)
+
+    minimum = max(
+        schema.get("minimum", default_min),
+        schema.get("exclusiveMinimum", default_min) + 1
+    )
+    maximum = min(
+        schema.get("maximum", default_max),
+        schema.get("exclusiveMaximum", default_max) - 1
+    )
+
+    not_multiple_of = schema.get("notMultipleOf", [])
+    if not isinstance(not_multiple_of, list):
+        not_multiple_of = [not_multiple_of]
+
+    for _ in range(MAX_REJECTED_SAMPLES):
+        # Generate new value
+        value = random_multiple_in_range(
+            minimum,
+            maximum,
+            multiple_of
+        )
+
+        # Verify
+        is_multiple_of = [value % num == 0 for num in not_multiple_of]
+        if any(is_multiple_of):
+            continue
+        return int(value)
+    raise RejectionSamplingFailed()
+
+
+def random_number(schema):
+    """Generate random number."""
+
+    minimum = schema.get("minimum", None)
+    exclusive_minimum = schema.get("exclusiveMinimum", None)
+    minimum = max(
+        minimum or Decimal("-Infinity"),
+        exclusive_minimum or Decimal("-Infinity")
+    )
+    if minimum == Decimal("-Infinity"):
+        minimum = None
+
+    maximum = schema.get("maximum", None)
+    exclusive_maximum = schema.get("exclusiveMaximum", None)
+    maximum = min(
+        maximum or Decimal("Infinity"),
+        exclusive_maximum or Decimal("Infinity")
+    )
+    if maximum == Decimal("Infinity"):
+        maximum = None
+
+    multiple_of = schema.get("multipleOf", None)
+
+    # Sample continuously if not given multiple_of
+    if not multiple_of:
+        if minimum and not maximum:
+            maximum = minimum + 100
+        elif maximum and not minimum:
+            minimum = maximum - 100
+        elif not minimum and not maximum:
+            # Use defualt range
+            minimum = -100
+            maximum = 100
+
+        # We don't have to worry about notMultipleOf
+        # because it's a continuous sample (infintesimal odds)
+        return Decimal(
+            random.uniform(float(minimum), float(maximum))
+        )
+
+    # Use multiple_of to sample
+    if minimum and not maximum:
+        given_range = multiple_of / minimum
```

Comment:
I don't follow this logic. More comments might help. When `minimum` is negative, do we end up with maximum<minimum?